### PR TITLE
Use MessageToDict in logging._gax

### PR DESCRIPTION
**Pull request by dhermes**
This allows removal of lots of custom conversion code. I noticed this when doing #2589.

### CODE REVIEW

**Comment by dhermes** [comment]
@tseaver FYI `ParseDict` handles the camel-case handling:

```python
>>> from google.protobuf.json_format import ParseDict
>>> from google.logging.type.http_request_pb2 import HttpRequest
>>> ParseDict({'request_url': 'a'}, HttpRequest())
request_url: "a"

>>> ParseDict({'requestUrl': 'a'}, HttpRequest())
request_url: "a"
```

**Comment by daspecster** [comment]
Here's a link to [`MessageToDict`](https://github.com/google/protobuf/blob/58580da37357941d502805be3ae520441be77728/python/google/protobuf/json_format.py#L112) just for easy reference.

File: logging/google/cloud/logging/_gax.py [hunk 1]
```diff
@@ -420,225 +419,14 @@ def metric_delete(self, project, metric_name):
...
-    if 'httpRequest' in mapping:
-        _http_request_mapping_to_pb(
-            mapping['httpRequest'], entry_pb.http_request)
-
-    if 'operation' in mapping:
-        _log_operation_mapping_to_pb(
-            mapping['operation'], entry_pb.operation)
-
+        mapping['timestamp'] = _datetime_to_rfc3339(mapping['timestamp'])
+    ParseDict(mapping, entry_pb)
```
Comment (tseaver) [inline logging/google/cloud/logging/_gax.py]: Does `ParseDict` actually know about the `multi_word_names` -> `multiWordNames` mappings?

File: logging/unit_tests/test__gax.py [hunk 2]
```diff
@@ -147,36 +170,74 @@ def test_list_entries_with_paging_nested_payload(self):
...
+
     def test_list_entries_with_extra_properties(self):
         from datetime import datetime
-        from google.logging.type.log_severity_pb2 import WARNING
-        from google.cloud._testing import _GAXPageIterator
+
+        # Import the wrappers to register the type URL for BoolValue
+        # pylint: disable=unused-variable
+        from google.protobuf import wrappers_pb2
+        # pylint: enable=unused-variable
```
Comment (tseaver) [inline logging/unit_tests/test__gax.py]: Grrr for import-time side-effects (I know, we commit this foul as well).

File: logging/google/cloud/logging/_gax.py (hunk 1 above)
Comment (dhermes) [inline logging/google/cloud/logging/_gax.py]: See my previous comment: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2600#issuecomment-255932028

**Comment by dhermes** [comment]
@tseaver Any remaining issues?

**Comment by tseaver** [comment]
LGTM

**Pull request merged**